### PR TITLE
fix: changes for CATS so they work for both Eirini and Diego

### PR DESCRIPTION
**Pull request by jimmykarily**
[WIP] The changes have been tested with the latest master of cf-acceptance-tests (our fork): https://github.com/SUSE/cf-acceptance-tests

but the release we use in the PR (0.0.20) doesn't include all the changes in master. There are 9 CATs still failing in Eirini on this branch. We should first fix them then create a new release for our cf-acceptance-tests-release and consume it in this PR to make it ready for review
 

## Checklist:

- [ ] My code has security implications.
- [x ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: chart/assets/operations/instance_groups/scheduler.yaml [hunk 1]
```diff
@@ -98,6 +98,14 @@
       fieldRef:
         fieldPath: status.podIP
 
+- type: replace
+  path: /instance_groups/name=scheduler/jobs/name=loggr-syslog-binding-cache/properties?/quarks
+  value:
+    ports:
+    - name: binding-cache
+      protocol: TCP
+      internal: 9000
```
Comment (jimmykarily) [inline chart/assets/operations/instance_groups/scheduler.yaml:107]: This change is fixing the syslog drain CATs. I wonder how those tests passed in diego.

File: mixins/eirini/templates/configmap.yaml [hunk 2]
```diff
@@ -95,4 +95,9 @@ data:
     eirini_cert_path: "/etc/eirini/secrets/eirini-client.crt"
     eirini_key_path: "/etc/eirini/secrets/eirini-client.key"
     ca_path: "/etc/eirini/secrets/eirini-client.ca"
+  instance-index-env-injector.yml: |
+    service_name: eirinix-instance-index-env-injector
+    service_port: 8443
+    workload_namespace: {{ .Values.eirini.opi.namespace }}
+    service_namespace: {{ .Release.Namespace }}
```
Comment (jimmykarily) [inline mixins/eirini/templates/configmap.yaml:102]: This change is adding the env injector webhook from the Eirini code (https://github.com/cloudfoundry-incubator/eirini/tree/master/cmd/instance-index-env-injector). It makes sure, the application pods get the CF_INSTANCE_INDEX env variable.

**Comment by jimmykarily** [comment]
last run from dk/cat_debug:

Summarizing 9 Failures:

[Fail] [apps] Getting instance information scaling memory [It] fails with insufficient resources 
/home/opensuse/cf-acceptance-tests/apps/instance_reporting.go:45

[Fail] [docker] Docker Application Lifecycle running a docker app with a start command [BeforeEach] retains its start command through starts and stops 
/home/opensuse/cf-acceptance-tests/docker/docker_lifecycle.go:52

[Fail] [docker] Docker Application Lifecycle running a docker app without a start command [BeforeEach] handles docker-defined metadata and environment variables correctly 
/home/opensuse/cf-acceptance-tests/docker/docker_lifecycle.go:79

[Fail] [docker] Docker Application Lifecycle running a docker app without a start command [BeforeEach] when env vars are set with 'cf set-env' prefers the env vars from cf set-env over those in the Dockerfile 
/home/opensuse/cf-acceptance-tests/docker/docker_lifecycle.go:79

[Fail] [v3] v3 docker app lifecycle [It] can run apps 
/home/opensuse/cf-acceptance-tests/helpers/v3_helpers/v3.go:478

[Fail] [apps] Crashing the app crashes [It] shows crash events 
/home/opensuse/cf-acceptance-tests/apps/crashing.go:65

[Fail] [apps] Application Lifecycle pushing [It] makes the app reachable via its bound route 
/home/opensuse/cf-acceptance-tests/apps/lifecycle.go:97

[Fail] [apps] Rolling deploys [It] deploys the app with zero downtime 
/home/opensuse/cf-acceptance-tests/apps/rolling_deploy.go:71

[Fail] [v3] buildpack With a single buildpack app [It] Downloads the correct user specified git buildpack 
/home/opensuse/cf-acceptance-tests/v3/buildpacks.go:88

Ran 97 of 211 Specs in 6621.271 seconds
FAIL! -- 88 Passed | 9 Failed | 3 Pending | 111 Skipped

**Comment by jimmykarily** [comment]
After running `cf enable-feature-flag diego_docker` the docker tests pass. We need to add that to our CI.

**Comment by jimmykarily** [comment]
The `crashing.go:65` is probably related to this: https://www.pivotaltracker.com/n/projects/2172361
It should be fixed in the next release of eirini. Will just Skip for now.

**Comment by viccuad** [comment]
I think that means we only fail:

```
[Fail] [apps] Rolling deploys [BeforeEach] deploys the app with zero downtime 
/home/vic/suse/cf-acceptance-tests/apps/rolling_deploy.go:27
```
(see https://github.com/cloudfoundry-incubator/kubecf/issues/1354)


and:

```
[Fail] [apps] Getting instance information scaling memory [It] fails with insufficient resources 
/home/vic/suse/cf-acceptance-tests/apps/instance_reporting.go:45
```

**Comment by viccuad** [comment]
For error:

```
[Fail] [apps] Getting instance information scaling memory [It] fails with insufficient resources 
/home/vic/suse/cf-acceptance-tests/apps/instance_reporting.go:45
```

It fails earlier in Eirini when doing a `cf scale binapp -m 99999G -f`  with `memory quota_exceeded` . If one tries `cf scale binapp -m 999G -f` if goes to start the instance and then fails with the expected `Error starting instances: 'Insufficient resources: memory'`  in a loop, the same as Diego.
Which I think means that in Eirini we just fail earlier, and we don't even try to instantiate if the mem quota is not enough.

This is the reproducer:
```
$ cf create-quota catnip-quota -m 99999G -i -1 -r 1000 -a -1 -s 100 --reserved-route-ports 20 --allow-paid-service-plans
$ cf create-org catnip-org
$ cf set-quota catnip-org catnip-quota
$ cf create-space -o catnip-org catnip-space
$ cf create-user catnip-user password
$ cf set-space-role catnip-user catnip-org catnip-space SpaceManager
$ cf set-space-role catnip-user catnip-org catnip-space SpaceAuditor
$ cf set-space-role catnip-user catnip-org catnip-space SpaceDeveloper
$ cf logout
$ cf auth catnip-user password
$ cf target -o catnip-org -s catnip-space
$ cf push binapp -p cf-acceptance-tests/assets/binary -b binary_buildpack -m 256M -c ./app
$ cf scale binapp -m 99999G -f
Scaling app binapp in org catnip-org / space catnip-space as catnip-user...
memory quota_exceeded
FAILED
Expected:
  Instances starting...
  Error starting instances: 'Insufficient resources: memory'
Yet this works:
$ cf scale binapp -m 999G
Scaling app binapp in org catnip-org / space catnip-space as catnip-user...
This will cause the app to restart. Are you sure you want to scale binapp? [yN]: y
Stopping app binapp in org catnip-org / space catnip-space as catnip-user...
Starting app binapp in org catnip-org / space catnip-space as catnip-user...
Instances starting...
Error starting instances: 'Insufficient resources: memory'
...
```

**Review by viovanov — APPROVED** [review]

**Pull request merged**